File: backend/src/main/java/com/bkav/device_mag_backend/model/entity/Device.java
package com.bkav.device_mag_backend.model.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Data
@Table(name="device")
public class Device {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "uuid DEFAULT gen_random_uuid()", updatable = false, nullable = false)
    private UUID deviceId;

    @Column(length = 100, nullable = false)
    private String name;

    @Column
    private String description;

    @Column(length = 20)
    @Enumerated(EnumType.STRING)
    private DeviceStatus status;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column
    private LocalDateTime updatedAt;

}

File: frontend/src/app/shared/model/device.ts
export interface Device {
    deviceId: string;
    name: string;
    description: string;
    status: DeviceStatus;
    createdAt: string;
    updatedAt: string;
}

export const DEVICE_KEYS: (keyof Device)[] = [
    'name',
    'description',
    'status',
    'createdAt',
    'updatedAt'
];

export enum DeviceStatus {
    AVAILABLE = "ACTIVE",
    RETURNING = "RETURNING",
    ASSIGNED = "ASSIGNED"
}

export const DEVICE_DISPLAYED_COLUMNS = {
    name: 'Tên thiết bị',
    description: 'Mô tả',
    status: 'Trạng thái',
    createdAt: 'Ngày tạo',
    updatedAt: 'Ngày cập nhật'
};

